src/lib/flash-helper.js

```diff
@@ -25,8 +27,9 @@ async function flashFiles({ device, flashSteps, resetAfterFlash = true, ui }) {
 				device = await _flashDeviceInDfuMode(device, step.data, { name: step.name, altSetting: altSetting, startAddr: step.address, progress: progress });
 			}
 		}
+		success = true;
 	} finally {
-		progress({ event: 'finish' });
+		progress({ event: 'finish', success });
```

Comment:
The UI was looking a bit weird as it says `Flash Success` and then `Operation failed due to device protection`.

So I am doing `Flash Failed`